### PR TITLE
TST: new test for incorrect series assignment

### LINKED ISSUE #14525: Inconsistent error raising in indexing
#### A small, complete example of the issue

``` python
x = np.random.randn(2, 2)
y = pd.Series(range(2))
y.loc[range(2)] = x # succeeds
y.loc[:] = x # fails
```
#### Expected Output

Since we're assigning a 2D array to a 1D vector that is smaller than the raveled array, I would expect both assignments to fail.

### PR DESCRIPTION
**Pull request by ganevgv**
- [x] closes #14525
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Review by mroeschke — APPROVED** [review]

File: pandas/tests/series/indexing/test_indexing.py [hunk 1]
```diff
@@ -391,6 +391,22 @@ def test_setslice(datetime_series):
     assert sl.index.is_unique is True
 
 
+def test_incorrect_assigment():
```
Comment (jreback) [inline pandas/tests/series/indexing/test_indexing.py]: can you change this test name to be more informative, something like

test_error_raised_on_2d_to_1d

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/series/indexing/test_indexing.py (hunk 1 above)
Comment (ganevgv) [reply pandas/tests/series/indexing/test_indexing.py]: Thanks for the feedback @jreback! Changed the test name to `test_2d_to_1d_assignment_error_raised`.

File: pandas/tests/series/indexing/test_indexing.py [hunk 2]
```diff
@@ -391,6 +391,22 @@ def test_setslice(datetime_series):
     assert sl.index.is_unique is True
 
 
+def test_2d_to_1d_assignment_error_raised():
```
Comment (simonjayhawkins) [inline pandas/tests/series/indexing/test_indexing.py]:
```suggestion
def test_2d_to_1d_assignment_raises():
```

**Review by simonjayhawkins — COMMENTED** [review]
@ganevgv minor nit. test name should end with `_raises` for consistency with other tests. otherwise lgtm.

File: pandas/tests/series/indexing/test_indexing.py (hunk 2 above)
Comment (ganevgv) [reply pandas/tests/series/indexing/test_indexing.py]: Thanks @simonjayhawkins - good to know

**Review by simonjayhawkins — APPROVED** [review]
@ganevgv lgtm @jreback

**Comment by mroeschke** [comment]
Thanks @ganevgv!

**Pull request merged**